### PR TITLE
fix(atomic): managed focus for breadbox show more/less

### PR DESCRIPTION
**Pull request by btaillon**
https://coveord.atlassian.net/browse/KIT-1724

image

### CODE REVIEW

**Comment by ThibodeauJF** [comment]
when doing show more/less the focus state is kind of weird
Screen Shot 2022-06-10 at 3 52 15 PM

vs the normal one
Screen Shot 2022-06-10 at 3 53 05 PM

**Comment by btaillon** [comment]
> when doing show more/less the focus state is kind of weird
> Screen Shot 2022-06-10 at 3 52 15 PM
> 
> vs the normal one
> Screen Shot 2022-06-10 at 3 53 05 PM
 
I'll fix that.

File: packages/atomic/src/utils/accessibility-utils.tsx [hunk 1]
```diff
@@ -75,8 +75,10 @@ export function FocusTarget() {
           element = el;
           if (focusOnNextTarget) {
             focusOnNextTarget = false;
-            element.focus();
-            onFocusCallback?.();
+            setTimeout(() => {
```
[the remark this answers is not in the document]
Comment (liviarett) [reply packages/atomic/src/utils/accessibility-utils.tsx:78]: would making this a self calling async func have the same effect while looking a bit less hacky, maybe?

[the remark this answers is not in the document]
Comment (olamothe) [reply packages/atomic/src/utils/accessibility-utils.tsx:78]: At the very least there should be a comment that explains why it's needed.

As for setTimeout 0 VS async function, maybe we could add a new function in `utils` named `delay` or `defer`. 

So a newcomer not that familiar with JS would understand better why it's like that.

And then how it's done in there (with a setTimeout 0 or async function) would not matter that much.

**Review by olamothe — APPROVED** [review]

**Review by ThibodeauJF — APPROVED** [review]

**Pull request merged**